### ISSUE TITLE
Incorrect results on GPU for reduce_* methods on large tensors

### ISSUE THREAD

**Issue by raphael-sch** [tensorflow/tensorflow#22123]
-----------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: NA
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.9.0 / 1.5.0 / 1.10.1
- **Python version**: 2.7.15 / 3.6.3
- **Bazel version (if compiling from source)**: NA
- **GCC/Compiler version (if compiling from source)**: NA
- **CUDA/cuDNN version**: CUDA 9.0
- **GPU model and memory**: K10, P100
- **Exact command to reproduce**: see code below

### Describe the problem
For tensors with large amount of values (2^28 seems to be the threshold) a few of the reduce_* (tested max, sum, min) methods don't work correctly. The incorrect values appear to be read randomly from memory. The issue was only observed on GPU. Other observations are that a tensor with int types works as expected. Also no problem if the tensor is read as tf.constant. 
These posts seem to be related:
https://github.com/tensorflow/tensorflow/issues/20094
https://stackoverflow.com/questions/49520394/tf-reduce-sum-on-gpu-fails-in-combination-with-placeholder-as-input-shape
https://stackoverflow.com/questions/50806742/tensorflow-reduce-sum-returns-partially-incorrect-output-given-a-large-input

### Source code / logs
https://colab.research.google.com/drive/1GNqzUfRTRRohyjiF8HqvWzvKE6LasQ-4

```
import tensorflow as tf
import numpy as np

n = np.random.randint(1,9)
inp = np.ones(shape=(140000000, 2)) * n
tensor = tf.placeholder(tf.float32, shape=(None, None))
output = tf.reduce_max(tensor, axis=-1)

with tf.Session() as sess:
    result = sess.run(output, feed_dict={tensor: inp})
    print('output values')
    print(result)
    print('number of incorrect values')
    print(np.sum(result != inp.max(axis=-1)))
```
Output:
```
output values
[5. 5. 5. ... 0. 0. 0.]
number of incorrect values
5782272
```

Pinging:
@nuance-research

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
OS Platform and Distribution
TensorFlow installed from
Bazel version
Mobile device

**Comment by raphael-sch** [comment]
updated

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by raphael-sch** [comment]
possible related: https://github.com/tensorflow/tensorflow/issues/19155

**Comment by bignamehyp** [comment]
@reedwm can you please take a look? Seems relevant.

**Comment by reedwm** [comment]
/CC @ekelsen, can you take a look?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ekelsen, @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by chsigg** [comment]
Sorry for the delay. I should have a fix for this shortly.

**Comment by chsigg** [comment]
This should be fixed in https://github.com/tensorflow/tensorflow/commit/fc44600e5c3ccf1de1e3d4792a00d3578311d3f6

I'm closing this issue, please let us know if you still experience this issue.
